### PR TITLE
Github Authentication

### PR DESCRIPTION
**Pull request by khalifan-kfan**
# Description
This feature is about using github auth to be able to login or sign up to crane cloud.
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Trello Ticket ID
https://trello.com/c/TCOm5uys

## How Can This Been Tested?
 checkout the branch and try to use github login instead.
you will need a client ID in you environment variable
Weather you registered  or not, you should be able to get in just by clicking login with github on the login page.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots

git button

### CODE REVIEW

File: src/components/LoginPage/index.js [hunk 1]
```diff
@@ -170,17 +222,38 @@ class LoginPage extends React.Component {
...
                 onClick={this.handleSubmit}
+              /> 
+            </div>
+          </form>
+          <div className="LowerLoginSection">
+          <div>
+                 <p className="LoginWith"><span>Or Login with</span></p>  
+               </div> 
+               <a
+                href={gitLoading ? 0 :`${GIT_REDIRECT_URL}`}
```
Comment (Mubangizi) [inline src/components/LoginPage/index.js:235]: Please remove this link and add an `onClick` listener to the button. One that will go to git auth

File: src/components/LoginPage/index.js [hunk 2]
```diff
@@ -21,20 +22,29 @@ class LoginPage extends React.Component {
       password: "",
       loading: false,
       error: "",
+      gitLoading: false,
+      feedbackMessage:"",
```
Comment (Mubangizi) [inline src/components/LoginPage/index.js:26]: There is no need for a new messaging element you can still use the previously existing feedback alerts

**Review by Mubangizi — COMMENTED** [review]
Hey @khalifan-kfan, great work. There is just a few changes to note

File: src/components/LoginPage/index.js (hunk 2 above)
Comment (khalifan-kfan) [reply src/components/LoginPage/index.js:26]: The thing was it was there before but  it wasn' put in state 
Which was a bug so i just added it to state

File: src/components/LoginPage/index.js (hunk 1 above)
Comment (khalifan-kfan) [reply src/components/LoginPage/index.js]: done

**Review by neelxie — APPROVED** [review]
lgtm

**Review by Mubangizi — APPROVED** [review]
LGTM

**Pull request merged**